x-pack/filebeat/input/gcppubsub/input.go

```diff
@@ -119,8 +117,14 @@ func NewInput(
 				for _, priv := range privates {
 					if msg, ok := priv.(*pubsub.Message); ok {
 						msg.Ack()
-						in.ackedCount.Inc()
+
+						now := time.Now()
+						in.metrics.ackedMessageCount.Inc()
+						in.metrics.bytesProcessedTotal.Add(uint64(len(msg.Data)))
+						in.metrics.processingTime.Update(now.Sub(msg.PublishTime).Nanoseconds())
```

Comment:
```suggestion
						in.metrics.processingTime.Update(time.Since(msg.PublishTime).Nanoseconds())
```
and remove the `now := time.Now()`.